job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -111,7 +113,13 @@ private boolean shouldIgnore(URIStatus status) {
 
     Set<Pair<WorkerInfo, ArrayList<CompactTask>>> result = Sets.newHashSet();
     for (Map.Entry<WorkerInfo, ArrayList<CompactTask>> assignment : assignments.entrySet()) {
-      result.add(new Pair<>(assignment.getKey(), assignment.getValue()));
+      List<List<CompactTask>> partitioned = CommonUtils.partition(
+          assignment.getValue(), JOBS_PER_WORKER);
+      for (List<CompactTask> compactTasks : partitioned) {
```

Comment:
can we add a simple unit test to see that `selectExecutors()` returns multiple tasks per host?